feat(datasetDownloadButton): implement "Download survey data" button

jalMogo opened the pull request:
Resolves: https://github.com/jalMogo/mgmt/issues/26

TODO's:
 - [x] depends on merging: https://github.com/mapseed/api/pull/140
 - [x] refactor references to the old AuthNavView in the flavor-specific AppView's

goldpbear commented on src/base/static/js/views/app-view.js:
```diff
@@ -221,6 +224,23 @@ module.exports = Backbone.View.extend({
+          href={`${apiRoot}${datasetOwner}/datasets/${datasetSlug}/mapseed-places.csv?format=csv&include_private=true&page_size=10000`}
+          style={{ float: "right" }}
+        >
+          <SecondaryButton>{`Download Survey Data`}</SecondaryButton>
```
I don't think the nested `SecondaryButton` is needed here-- I think the `Link` component by itself should be enough.

jalMogo replied:
I suppose a link looks a bit cleaner. This is how it currently looks with just the link:
![download-data-header](https://user-images.githubusercontent.com/42042460/44185216-d78bce80-a0c7-11e8-8802-136af6981970.png)

in re:
>An alternative UI might nest the download button in the flyout that appears when you click on a logged-in user avatar. 

I think that is preferable. Downloading the dataset is an infrequently used feature, so it only clutters the limited mobile space by keeping it up there.

I may need to port the PagesNavView view to React, so I'll explore whether that is worth doing.

jalMogo commented on src/base/static/components/molecules/user-menu.js:
```diff
@@ -0,0 +1,126 @@
+            })}
+          >
+            {isAdmin && (
+              <li style={{ paddingBottom: "16px" }}>
```
I'm not too excited about the inline styles here and the confusing state of our CSS.

After this PR is merged, I think I'll take a pass on this component using CSS-in-JS and see if it cleans things up.

goldpbear replied:
Do we want to move the CSS used here into a `user-menu.scss` file and apply BEM styles? That's how we've handled other components ports so far.

jalMogo commented on src/base/static/js/views/pages-nav-view.js:
```diff
@@ -50,8 +49,4 @@ module.exports = Backbone.View.extend({
-  onAuthNavClick: function(evt) {
-    evt.preventDefault();
-    $(".sign-in-menu").toggleClass("is-exposed");
-  },
```
This code was not being used - it seems irrelevant to PagesNavView, because it is specific to AuthNavView. Perhaps it was left over from a copy/paste operation?

Either way, it was confusing and unused, so I removed it.

jalMogo commented on webpack.config.js:
```diff
@@ -156,5 +156,6 @@ module.exports = {
     },
     compress: true,
     port: PORT,
+    allowedHosts: [".ngrok.io"],
```
This was needed for testing with ngrok.

goldpbear commented on src/base/static/components/molecules/user-menu.js:
```diff
@@ -0,0 +1,126 @@
+                    this.props.apiRoot
+                  }${datasetOwner}/datasets/${datasetSlug}/mapseed-places.csv?format=csv&include_private=true&page_size=10000`}
+                  classes={".btn-secondary"}
+                  style={{ width: "100%", textDecoration: "none" }}
```
Ideally we shouldn't have to pass through legacy CSS classes like `btn-secondary`. The `Link` component should have data-driven styles in it that replicate the styling in the legacy scss files.

I think if you pass a `color="secondary"` prop to a `Link` it will adopt the style of the Sign-In button.

goldpbear commented on src/base/static/components/molecules/user-menu.js:
```diff
@@ -0,0 +1,126 @@
+
+  render() {
+    const {
+      slug: datasetSlug,
```
If I run this code on a flavor without a `dataset_download` section in the config, I get an error:
```
user-menu.js:82 Uncaught TypeError: Cannot read property 'slug' of undefined
```
Maybe we should have a default value for the `datasetDownloadConfig` of an empty object.

jalMogo commented:
@goldpbear I've made the following updates:
 - fixed the bug when a `dataset_download` config is not present on the flavor
 - updated august, bathgate, hull, manzanares, and williams flavors to reference the new UserMenu component. All references to AuthNavView should now be gone.

I'm not planning to make any of the CSS changes in the PR - I am working on a CSS-in-JS implementation that will clean up the CSS in UserMenu, as well as add support for theming. If that's alright with you, maybe we can merge this and I'll submit a followup PR within a couple hours.

goldpbear commented:
Sounds good-- I think it's fine to wait on the CSS changes.

I'll go ahead and close https://github.com/mapseed/platform/pull/992 since it looks like the VV flavors got addressed in this PR.
